R/validate-config-utils.R

```diff
@@ -478,6 +478,56 @@ validate_mt_property_unique_vals <- function(model_task_grp,
   }
 }
 
+# Check that modeling task round ids match the expected round ID patterns when
+# round_id_from_variable = TRUE
+validate_mt_round_id_pattern <- function(model_task_grp,
+                                         model_task_i,
+                                         round_i,
```

Comment:
I see that these appear in function definitions throughout this file, but I note that they're not used within this function.  It's not clear to a new reader of this code why these are needed here.  I guess you're just aiming for a consistent API so you can use `imap` everywhere below?

Reply:
They are used in `get_error_paths()` but do not need to be passed explicitly as they are sourced by the caller environment. They are used to create the indices to specific instances of rounds, model tasks, target metadata etc

Reply:
Actually, let me take a closer look!

Reply:
OK yes I've confirmed it and clarified the `get_error_path` documentation. It's described in the documentation of the function which returns non interpolated glue formatted paths (e.g. `"/rounds/{round_i - 1}/model_tasks/{model_task_i - 1}/task_ids/origin_date"` which then get's interpolated when evaluated by `glue` to become: `"/rounds/1/model_tasks/0/task_ids/origin_date"`.

I added some additional clarifications and some explicit demonstration in some additional tests in 1a7019d1c306cea6603c0d3f2d7b4c5962da252d

While the get_error_path() docs demonstrate and discuss this it seems that it might be good to add a note somewhere else too. However, I'm not sure where. It's behaviour that's used throughout so I don't want to have to comment every single instance. Happy for suggestions!

Reply:
Thanks for the explanation!  I also can't think of a good other place to add more about this.  Seems good enough to me.